packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/util.ts

```diff
@@ -38,3 +37,36 @@ async function waitForStatementComplete(statementId: string): Promise<void> {
 export function makePhysicalId(resourceName: string, clusterProps: ClusterProps, requestId: string): string {
   return `${clusterProps.clusterName}:${clusterProps.databaseName}:${resourceName}:${requestId}`;
 }
+
+export function getDistKeyColumns(columns: Column[]): Column[] {
+  // string comparison is required for custom resource since everything is passed as string
+  return columns.filter(column => column.distKey === true || (column.distKey as unknown as string) === 'true');
+}
+
+export function getDistKeyColumn(columns: Column[]): Column | undefined {
+  const distKeyColumns = getDistKeyColumns(columns);
+  if (distKeyColumns.length === 0) {
+    return undefined;
+  } else if (distKeyColumns.length > 1) {
+    throw new Error('Multiple dist key columns found');
+  }
```

Comment:
It seems like this logic belongs in getDistKeyColumns itself; or rather, that we only need `getDistKeyColumn`.